### PR TITLE
[types] Define rooch-types and use H256 to replace HashValue

### PR DESCRIPTION
**Pull request by jolestar**
1. Define rooch-types and draft the rooch transaction types.
2. Migrate the moveos::types to moveos-types.
3. Use H256 to replace HashValue.

TODO
1. How to support multi chain transaction
2. Transaction validation

### CODE REVIEW

File: crates/rooch-types/src/transaction/mod.rs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// Copyright (c) RoochNetwork
+// SPDX-License-Identifier: Apache-2.0
+
+use ethereum::{EthereumSignature, EthereumTransactionData};
+
+pub mod ethereum;
+
+pub enum TransactionType {
+    Ethereum,
+    Move,
```
Comment (baichuan3) [inline crates/rooch-types/src/transaction/mod.rs:10]: Need to distinguish transactions from different Move chains?

File: moveos/moveos-types/src/h256.rs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// Copyright (c) RoochNetwork
+// SPDX-License-Identifier: Apache-2.0
+
+pub use ethereum_types::H256;
+use tiny_keccak::{Hasher, Sha3};
+
+pub const LENGTH: usize = 32;
```
Comment (baichuan3) [inline moveos/moveos-types/src/h256.rs:7]: The difference of this method is hash[.32], the old method is hash[..], is it for compatibility with the Ethereum chain?

Comment (jolestar) [reply moveos/moveos-types/src/h256.rs:7]: Old HashValue also hash[.32].

File: crates/rooch-types/src/transaction/mod.rs (hunk 1 above)
Comment (jolestar) [reply crates/rooch-types/src/transaction/mod.rs:10]: Just use an example for support multichain transactions, it is not finished yet.

**Review by baichuan3 — APPROVED** [review]

**Comment by jolestar** [comment]
part of #72

**Pull request merged**